casper/src/main/scala/coop/rchain/casper/BlockCreator.scala

```diff
@@ -73,12 +73,10 @@ object BlockCreator {
         invalidBlocksSet <- dag.invalidBlocks
         invalidBlocks    = invalidBlocksSet.map(block => (block.blockHash, block.sender)).toMap
         // make sure closeBlock is the last system Deploy
-        systemDeploys = if (slashingDeploys.nonEmpty)
-          slashingDeploys ++ Seq(
-            CloseBlockDeploy(Tools.rng(parents.head.blockHash.toByteArray))
-          )
-        else List.empty[SystemDeploy]
-        unsignedBlock <- if (deploys.nonEmpty || systemDeploys.nonEmpty) {
+        systemDeploys = slashingDeploys :+ CloseBlockDeploy(
+          Tools.rng(parents.head.blockHash.toByteArray)
+        )
+        unsignedBlock <- if (deploys.nonEmpty || slashingDeploys.nonEmpty) {
```

Comment:
There is something wrong with closeBlock when merging now. I think we should leave this the old way now.

Reply:
Close block is always in the list of `systemDeploys` so if there is no user `deploys`, block is created with only close block deploy, and error is throw that block is empty. I'm not sure if slashing deploy can be alone in the block.

Reply:
> Close block is always in the list of `systemDeploys` so if there is no user `deploys`, block is created with only close block deploy, and error is throw that block is empty. I'm not sure if slashing deploy can be alone in the block.

Ah. You are right. 

> slashing deploy can be alone in the block

I am not sure what you mean about this sentence. 

BTW, we should check if the slashing deploy would cause merge conflicts also.

Reply:
This condition `if (deploys.nonEmpty || slashingDeploys.nonEmpty)` allows block to be created without user deploys, slashing deploys are checked like user deploys.

Reply:
Old way is fine with me. It seems to be working properly.